apollo-router-core/src/query_planner/mod.rs

```diff
@@ -323,14 +320,16 @@ mod fetch {
 
             let query_span = tracing::info_span!("subfetch", service = service_name.as_str());
 
-            let Variables { variables, paths } = Variables::new(
-                &self.requires,
-                self.variable_usages.as_ref(),
-                data,
-                current_dir,
-                request,
-                schema,
-            )?;
+            let Variables { variables, paths } = query_span.in_scope(|| {
```

Comment:
:yarn: per our ADR on telemetry, we should prefer to put this at the source instead of the caller.